src/linker/Linker/WarningSuppressor.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class WarningSuppressor
```

Comment:
Nit: I don't like the name - this in itself doesn't suppress warnings. Maybe `WarningSuppressionWriter`.